independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InjectionPointInfo.java

```diff
@@ -43,30 +43,35 @@
  */
 public class InjectionPointInfo {
 
-    static InjectionPointInfo fromField(FieldInfo field, ClassInfo beanClass, BeanDeployment beanDeployment) {
+    static InjectionPointInfo fromField(FieldInfo field, ClassInfo beanClass, BeanDeployment beanDeployment,
+            InjectionPointModifier transformer) {
         Set<AnnotationInstance> qualifiers = new HashSet<>();
         for (AnnotationInstance annotation : beanDeployment.getAnnotations(field)) {
             if (beanDeployment.getQualifier(annotation.name()) != null) {
                 qualifiers.add(annotation);
             }
         }
-        return new InjectionPointInfo(
-                resolveType(field.type(), beanClass, field.declaringClass(), beanDeployment),
-                qualifiers.isEmpty() ? Collections.emptySet() : qualifiers, field, -1);
+        Type type = resolveType(field.type(), beanClass, field.declaringClass(), beanDeployment);
+        return new InjectionPointInfo(type,
+                transformer == null ? qualifiers : transformer.applyTransformers(type, field, qualifiers), field, -1);
```

Comment:
`transformer` cannot be `null`, or?

Reply:
In practice, not.
In tests, there is `BeanInfoTypesTest#testResolver` and `BeanInfoQualifiersTest#testQualifiers` which both create `BeanInfo bean = Beans.createClassBean(...)` with `null` for tranformer param.
At some point I had test failures due to this but not anymore so technically it's redundant now - want me to remove that?

Reply:
I just went ahead and removed those checks.